### PR TITLE
2024-05-06 update chatter-to-service-form-view

### PR DESCRIPTION
**Pull request by nirmalsankalpa**
2024-05-06 update chatter-to-service-form-view
image

### CODE REVIEW

File: custom_crm_client/models/services.py [hunk 1]
```diff
@@ -15,12 +16,29 @@ class Services(models.Model):
...
     tags = fields.Char('Tags')
     contact = fields.Char('Contact')
+    startdate = fields.Date(string='Start Date')
     enddate = fields.Date(string='End date')
 
-    stage_name = fields.Many2one('crm.service.stage',  string='stage_name')
+
+    #stage_name = fields.One2many('crm.client', 'service_id', string='Stage Name')
+
+    stage_ID = fields.Many2one('crm.service_stage',
```
[the remark this answers is not in the document]
Comment (nirmalsankalpa) [reply custom_crm_client/models/services.py:25]: I am trying to use this stage_ID in the kanban view

File: custom_crm_client/models/services.py [hunk 2]
```diff
@@ -15,12 +16,29 @@ class Services(models.Model):
...
+                                string='Stage Name',
+                                ondelete='set null',
+                                help="Select a Stage.")
+    uci_no = fields.Integer(string='UCI Number ')
 
     client_id = fields.Many2one('crm.client',  string='client_id')
 
+    serviceslocation = fields.Selection([
```
[the remark this answers is not in the document]
Comment (nirmalsankalpa) [reply custom_crm_client/models/services.py:33]: done

File: custom_crm_client/views/services.xml [hunk 3]
```diff
@@ -16,28 +16,53 @@
...
                     <group>
                         <field name="serviceTypes" />
 <!--                        <field name="servicestage"  />-->
+<!--                        <field name="stage_ID"-->
+<!--                                   options="{'no_create': True, 'no_quick_create': True, 'no_create_edit': True}"-->
+<!--                                   widget="many2one" can_create="false" can_write="true"/>-->
                         <field name="responsible"/>
                         <field name="tags"/>
                         <field name="contact"/>
+                        <field name="serviceslocation"/>
```
[the remark this answers is not in the document]
Comment (nirmalsankalpa) [reply custom_crm_client/views/services.xml:37]: done

File: custom_crm_client/models/services.py [hunk 4]
```diff
@@ -15,12 +16,29 @@ class Services(models.Model):
...
 
-    stage_name = fields.Many2one('crm.service.stage',  string='stage_name')
+
+    #stage_name = fields.One2many('crm.client', 'service_id', string='Stage Name')
+
+    stage_ID = fields.Many2one('crm.service_stage',
+                                string='Stage Name',
+                                ondelete='set null',
+                                help="Select a Stage.")
+    uci_no = fields.Integer(string='UCI Number ')
```
[the remark this answers is not in the document]
Comment (nirmalsankalpa) [reply custom_crm_client/models/services.py:29]: done

File: custom_crm_client/views/services.xml [hunk 5]
```diff
@@ -16,28 +16,53 @@
...
                         <group>
-                            <field name="date"/>to<field name="enddate"/>
-                            <field name="serviceTypes" />
+
+                            <field name="service_id"/>
+                             <field name="date"/>
+                             <field name="startdate"/>
+                            <field name="enddate"/>
                             <field name="description" placeholder=" Enter Your Description"/>
+                            <field name="ircc_servicestype" />
```
[the remark this answers is not in the document]
Comment (nirmalsankalpa) [reply custom_crm_client/views/services.xml:25]: done

File: custom_crm_client/models/services.py [hunk 6]
```diff
@@ -15,12 +16,29 @@ class Services(models.Model):
...
 
     client_id = fields.Many2one('crm.client',  string='client_id')
 
+    serviceslocation = fields.Selection([
+        ('Mcaf', 'Mcaf'),
+        ], string=" Service Location")
+
+    ircc_servicestype = fields.Selection([
```
[the remark this answers is not in the document]
Comment (nirmalsankalpa) [reply custom_crm_client/models/services.py:37]: done

File: custom_crm_client/models/service_stage.py [hunk 7]
```diff
@@ -1,9 +1,20 @@
-from odoo import models, fields
+import re
```
[the remark this answers is not in the document]
Comment (nirmalsankalpa) [reply custom_crm_client/models/service_stage.py:1]: done

File: custom_crm_client/views/services.xml [hunk 8]
```diff
@@ -48,8 +73,10 @@
         <field name="arch" type="xml">
              <tree>
                     <field name="name"/>
-                    <field name="description"/>
                     <field name="date"/>
+                    <field name="description"/>
+
+                    <field name="startdate"/><field name="enddate"/>
                     <field name="serviceTypes" widget="handle"/>
```
[the remark this answers is not in the document]
Comment (nirmalsankalpa) [reply custom_crm_client/views/services.xml]: done

File: custom_crm_client/models/services.py [hunk 9]
```diff
@@ -4,6 +4,7 @@ class Services(models.Model):
     _name = 'crm.services'
     _description = "CRM Services"
    # _inherits = {'res.partner': 'partner_id'}
```
Comment (RajatMCAF) [inline custom_crm_client/models/services.py:6]: Do we really need this comment?

**Review by RajatMCAF — COMMENTED** [review]
Review comments

File: custom_crm_client/models/services.py (hunk 9 above)
Comment (nirmalsankalpa) [reply custom_crm_client/models/services.py]: done

**Review by AndriiK139MCAF — APPROVED** [review]
Evrything seems good

**Pull request merged**